### PR TITLE
[Enhancement][Lake] Compaction scheduling based on compaction score (3/3)

### PR DESCRIPTION
**Pull request by sduzh**
This PR mainly to give users better insight into the scheduling of compaction tasks:
- Show compaction scores in the output of command `show partitions from table`, if the table is a lake table
- Add `show proc /compactions` to show current running compaction tasks

Example output of `show partitions from table`:
```
+-------------+---------------+----------------+----------------+-------------+--------+--------------+-------------------------------------------------------------------------+-----------------+---------+----------+----------+----------+------------+-------+-------+-------+
| PartitionId | PartitionName | CompactVersion | VisibleVersion | NextVersion | State  | PartitionKey | Range                                                                   | DistributionKey | Buckets | DataSize | RowCount | CacheTTL | AsyncWrite | AvgCS | P50CS | MaxCS |
+-------------+---------------+----------------+----------------+-------------+--------+--------------+-------------------------------------------------------------------------+-----------------+---------+----------+----------+----------+------------+-------+-------+-------+
| 10444       | p1            | 1              | 101            | 102         | NORMAL | lo_orderdate | [types: [INT]; keys: [-2147483648]; ..types: [INT]; keys: [19930101]; ) | lo_orderkey     | 48      | 2.2GB    | 91248844 | 0        | false      | 0.00  | 0.00  | 0.00  |
| 10445       | p2            | 1              | 101            | 102         | NORMAL | lo_orderdate | [types: [INT]; keys: [19930101]; ..types: [INT]; keys: [19940101]; )    | lo_orderkey     | 48      | 2.2GB    | 91007488 | 0        | false      | 0.00  | 0.00  | 0.00  |
| 10446       | p3            | 1              | 101            | 102         | NORMAL | lo_orderdate | [types: [INT]; keys: [19940101]; ..types: [INT]; keys: [19950101]; )    | lo_orderkey     | 48      | 2.2GB    | 91044214 | 0        | false      | 0.00  | 0.00  | 0.00  |
| 10447       | p4            | 1              | 101            | 102         | NORMAL | lo_orderdate | [types: [INT]; keys: [19950101]; ..types: [INT]; keys: [19960101]; )    | lo_orderkey     | 48      | 2.2GB    | 91016436 | 0        | false      | 0.00  | 0.00  | 0.00  |
| 10448       | p5            | 1              | 101            | 102         | NORMAL | lo_orderdate | [types: [INT]; keys: [19960101]; ..types: [INT]; keys: [19970101]; )    | lo_orderkey     | 48      | 2.2GB    | 91301792 | 0        | false      | 0.00  | 0.00  | 0.00  |
| 10449       | p6            | 1              | 101            | 102         | NORMAL | lo_orderdate | [types: [INT]; keys: [19970101]; ..types: [INT]; keys: [19980101]; )    | lo_orderkey     | 48      | 2.2GB    | 91050840 | 0        | false      | 0.00  | 0.00  | 0.00  |
| 10450       | p7            | 1              | 101            | 102         | NORMAL | lo_orderdate | [types: [INT]; keys: [19980101]; ..types: [INT]; keys: [19990101]; )    | lo_orderkey     | 48      | 1.3GB    | 53368288 | 0        | false      | 0.00  | 0.00  | 0.00  |
+-------------+---------------+----------------+----------------+-------------+--------+--------------+-------------------------------------------------------------------------+-----------------+---------+----------+----------+----------+------------+-------+-------+-------+
```
`AvgCS` is the average compaction score of the partition;
`P50CS` is the 50th percentile compaction score of the partition;
`MaxCS` is the maximum compaction score of the partition.

Example output of `show proc /compactions`:
```
StarRocks > show proc "/compactions";
+----------------------------+-------+---------------------+
| Partition                  | TxnID | StartTime           |
+----------------------------+-------+---------------------+
| ssb_lake_100g.lineorder.p3 | 25    | 2022-11-25 11:56:05 |
+----------------------------+-------+---------------------+
```

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

- [ ] I have checked the version labels which the pr will be auto backported to target branch

### CODE REVIEW

**Review by tracymacding — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java [hunk 1]
```diff
@@ -68,38 +71,53 @@ public class PartitionsProcDir implements ProcDirInterface {
...
-
-        if (this.partitionType == PartitionType.LIST) {
-            builder.add("List");
-        } else {
-            builder.add("Range");
-        }
-
-        builder.add("DistributionKey").add("Buckets").add("ReplicationNum");
-        if (olapTable.isLakeTable()) {
-            builder.add("EnableStorageCache").add("StorageCacheTtlSecond").add("AllowAsyncWriteBack")
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:96]: why remove cache and async write back properties?

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java [hunk 2]
```diff
@@ -68,38 +71,53 @@ public class PartitionsProcDir implements ProcDirInterface {
...
+                    .add("PartitionId")
+                    .add("PartitionName")
+                    .add("VisibleVersion")
+                    .add("VisibleVersionTime")
+                    .add("VisibleVersionHash")
+                    .add("State")
+                    .add("PartitionKey")
+                    .add(partitionType == PartitionType.LIST ? "List" : "Range")
+                    .add("DistributionKey")
+                    .add("Buckets")
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:111]: these properties are also needed in lake table.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:111]: These properties are fixed properties and can be fetched from `show create table`, I removed these properties for Keeps the output width less than the screen width.

**Comment by gengjun-git** [comment]
`This pr will affect users' behaviors`. You should mark this in pr checkbox. And more, starrocks manager should make some change to recognize these columns because it uses `versionCount` of the result of `show partitions`.

**Comment by sduzh** [comment]
> 

This PR will only change the behavior of the lake table and the lake table has not been released yet.

**Comment by sduzh** [comment]
> 



> `This pr will affect users' behaviors`. You should mark this in pr checkbox. And more, starrocks manager should make some change to recognize these columns because it uses `versionCount` of the result of `show partitions`.

For OlapTable, the output of `show partitions` remains unchanged.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java (hunk 2 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:111]: what does fixed properties mean?
partition key and buckets(maybe) are different between partitions.

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java (hunk 2 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:111]: i think whether these properties are required is more important than `Keeps the output width less than the screen width`.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java [hunk 3]
```diff
@@ -68,38 +71,53 @@ public class PartitionsProcDir implements ProcDirInterface {
...
-        if (olapTable.isLakeTable()) {
-            builder.add("EnableStorageCache").add("StorageCacheTtlSecond").add("AllowAsyncWriteBack")
-                    .add("DataSize").add("RowCount");
+        if (table.isLakeTable()) {
+            ImmutableList.Builder<String> builder = new ImmutableList.Builder<String>()
+                    .add("PartitionId")
+                    .add("PartitionName")
+                    .add("CompactVersion")
+                    .add("VisibleVersion")
+                    .add("NextVersion")
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:92]: What's the meaning of CompactionVersion and NextVersion?

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java [hunk 4]
```diff
@@ -68,38 +71,53 @@ public class PartitionsProcDir implements ProcDirInterface {
...
+                    .add("State")
+                    .add("DataSize")
+                    .add("RowCount")
+                    .add("AvgCS") // Average compaction score
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:96]: If you change the output, it's better to align it with our own table. Two different outputs make users misunderstand.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 73 / 141 (51.77%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CompactionsProcDir.java|6|22|27.27%|[27, 32, 37, 38, 39, 40, 41, 43, 44, 45, 46, 47, 48, 50, 51, 52]|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|66|118|55.93%|[87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 280, 281, 282, 285, 286, 289, 293, 294, 295, 296, 297, 298, 299, 300, 303, 304, 334, 340, 362, 363, 364, 366, 368, 369, 370, 371, 372, 373, 375, 376, 377, 378, 379, 380, 381, 382, 422, 429, 434]|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:96]: fixed, PTAL

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java (hunk 2 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:118]: Fixed, PTAL

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 72 / 151 (47.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CompactionsProcDir.java|6|22|27.27%|[27, 32, 37, 38, 39, 40, 41, 43, 44, 45, 46, 47, 48, 50, 51, 52]|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|65|128|50.78%|[88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 287, 288, 289, 292, 293, 296, 300, 301, 302, 303, 304, 305, 306, 307, 310, 311, 326, 332, 369, 370, 371, 372, 373, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 432, 439, 444]|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        BaseProcResult result = new BaseProcResult();
+        result.setNames(titles);
+        CompactionManager compactionManager = GlobalStateMgr.getCurrentState().getCompactionManager();
+        if (compactionManager == null) {
+            return result;
+        }
+        ConcurrentHashMap<PartitionIdentifier, CompactionContext> runningCompactions = compactionManager.getRunningCompactions();
+        for (CompactionContext context : runningCompactions.values()) {
+            List<String> row = new ArrayList<>();
+            row.add(String.valueOf(context.getPartitionName()));
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java:46]: Better to add table id or table name.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import com.starrocks.lake.compaction.CompactionContext;
+import com.starrocks.lake.compaction.CompactionManager;
+import com.starrocks.lake.compaction.PartitionIdentifier;
+import com.starrocks.server.GlobalStateMgr;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CompactionsProcDir implements ProcDirInterface {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java:16]: Better to implement `ProcNodeInterface` if there is no next level and no need to lookup.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java (hunk 5 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java]: `context.getPartitionName()` contains database and table name.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java (hunk 6 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/CompactionsProcDir.java]: Fixed, PTAL

**Review by wyb — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 72 / 150 (48.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionScheduler.java|0|1|00.00%|[249]|
|:large_blue_circle:|com/starrocks/common/proc/CompactionsProcNode.java|6|20|30.00%|[27, 28, 29, 30, 31, 33, 34, 35, 36, 37, 38, 40, 41, 42]|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|65|128|50.78%|[88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 287, 288, 289, 292, 293, 296, 300, 301, 302, 303, 304, 305, 306, 307, 310, 311, 326, 332, 369, 370, 371, 372, 373, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 432, 439, 444]|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**